sql/mysqld.cc

```diff
@@ -9597,7 +9598,10 @@ static void set_server_version(void)
     end= strmov(end, "-debug");
 #endif
   if (opt_log || opt_slow_log || opt_bin_log)
-    strmov(end, "-log");                        // This may slow down system
+    end= strmov(end, "-log");                        // This may slow down system
+
+  DBUG_ASSERT(end <= server_version + SERVER_VERSION_LENGTH);
```

Comment:
Shouldn't this assertion use just a <?

strmov/strxmov return a pointer to the terminating null character, so if it equals, it's already a one byte overflow.

Reply:
Fixed, thanks